runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1606,7 +1608,41 @@ MM_CopyForwardScheme::copyForwardCollectionSet(MM_EnvironmentVLHGC *env)
 	/* Do any final work to regions in order to release them back to the main collector implementation */
 	postProcessRegions(env);
 
-	return copyForwardCompletedSuccessfully(env);
+	static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_abortFlagRaisedDuringPGC = copyForwardCompletedSuccessfully(env);
+}
+
+#if defined(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD)
+void
+MM_CopyForwardScheme::concurrentCopyForwardCollectionSet(MM_EnvironmentVLHGC *env)
+{
+	/* isConcurrentCycleInProgress() tells us if this is the first PGC increment or not */
+	if (!isConcurrentCycleInProgress())
+	{
+		copyForwardPreProcess(env);
+	}
+
+	/* And perform the copy forward. Temporary as this will be updated to call concurrent copy forward state machine */
+	MM_CopyForwardSchemeTask copyForwardTask(env, _dispatcher, this, env->_cycleState);
+	_dispatcher->run(env, &copyForwardTask);
+
+	/* isConcurrentCycleInProgress() tells us if this is the last PGC increment or not */
+	if (!isConcurrentCycleInProgress())
+	{
+		copyForwardPostProcess(env);
+	}
+}
+#endif /* defined(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD) */
+
+void
+MM_CopyForwardScheme::copyForwardCollectionSet(MM_EnvironmentVLHGC *env)
```

Comment:
This method is effectively redunant - it is just one scenario of concurrentCopyForwardCollectionSet, where isConcurrentCycleInProgress is always false (but in future, the middle step of that method will need to distinguish further what Task/subTask needs to be done).

For clarity we could keep this - up to you.

Reply:
We'll soon be updating `concurrentCopyForwardCollectionSet` to call the copy forward state machine instead of the following:
```
	MM_CopyForwardSchemeTask copyForwardTask(env, _dispatcher, this, env->_cycleState);
	_dispatcher->run(env, &copyForwardTask);
```
With that in mind I think it's better to keep both versions